go/saltpack/errors.go

```diff
@@ -47,6 +47,23 @@ var (
 
 	// ErrBadLookup is when the user-provided key lookup gives a bad value
 	ErrBadLookup = errors.New("bad key lookup")
+
+	// ErrNoStream is returned if a stream constructor returned a nil
+	// stream.
+	ErrNoStream = errors.New("no stream")
+
+	// ErrHeaderNotRead is returned if a stream is read before the header is processed.
+	ErrHeaderNotRead = errors.New("message read before header processed")
```

Comment:
I didn't wind up needing this (see the comment above about not needing the `header` state).